src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,125 @@
+            switch (Enum.valueOf(Types.class, input[0].toUpperCase())) {
+            case BYE:
+                ui.printGoodbyeMessage();
+                return true;
+            case LIST:
+                ui.printTasksMessage();
+                taskList.printTasks();
+                return false;
+            case MARK: {
+                try {
+                    int taskToMark = Integer.parseInt(input[1]) - 1;
+                    Task currTask = taskList.get(taskToMark);
+                    currTask.setAsDone();
+                    ui.printMarkedTaskMessage(currTask.toString());
+                } catch (IndexOutOfBoundsException e) {
+                    throw new NeroException("Please add the correct index from 0 to "
+                            + taskList.getSize());
+                }
+                return false;
+            }
+            case UNMARK: {
+                try {
+                    int taskToUnmark = Integer.parseInt(input[1]) - 1;
+                    Task currTask = taskList.get(taskToUnmark);
+                    currTask.setAsUndone();
+                    ui.printUnmarkedTaskMessage(currTask.toString());
+                } catch (IndexOutOfBoundsException e) {
+                    throw new NeroException("Please add the correct index from 0 to "
+                            + taskList.getSize());
+                }
+                return false;
+            }
+            case TODO:
+                int index = originalString.indexOf("todo");
+                try {
+                    String toAdd = originalString.substring(index + 5);
+                    Task newTask = new ToDo(toAdd);
+                    taskList.addTask(newTask);
+                    ui.printAddedTasks(newTask.toString(), taskList.getSize());
+                } catch (IndexOutOfBoundsException e) {
+                    throw new NeroException("Description cannot be empty!!!");
+                }
+                return false;
+            case DEADLINE:
+                try {
+                    String[] splitString = originalString.split("/");
+                    String description = splitString[0].replace("deadline", "");
+                    String duration = splitString[1].replace("by", "");
+                    Task newTask = new Deadline(description, duration);
+                    taskList.addTask(newTask);
+                    ui.printAddedTasks(newTask.toString(), taskList.getSize());
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Add a task description and deadline!!!");
+                }
+                return false;
+            case EVENT:
+                try {
+                    String[] splitString = originalString.split("/");
+                    String description = splitString[0].replace("event", "");
+                    String startDate = splitString[1].replace("from", "");
+                    String endDate = splitString[2].replace("to", "");
```

Comment:
Would it be better to abstract out the initial handling of string inputs from users?

Reply:
Yep sounds good!